opensquirrel/utils/matrix_expander.py

```diff
@@ -41,12 +51,13 @@ def get_reduced_ket(ket: int, qubits: Iterable[Qubit]) -> int:
     """
     reduced_ket = 0
     for i, qubit in enumerate(qubits):
+        qubit = Qubit(qubit)
         reduced_ket |= ((ket & (1 << qubit.index)) >> qubit.index) << i
```

Comment:
Maybe `&` and `<<` and `>>` operations can be overloaded for `Qubit`, so you don't need the line `qubit = Qubit(qubit)`.

I.e., you could be able to use a qubit as an int.

Reply:
I will make a ticket for this as I think this is out of scope and would require us to investigate whether we would want this functionality or not

Reply:
It is usually good practice in Python to allow your function inputs to be as broad as possible and your return to be as specific as possible. I fear that by doing this, we would force Qubit objects onto `get_reduced_ket`. 

I think a ticket is good, this is not the only function where this functionality should be instilled. I think this conversion should be part of the IR itself.

Reply:
> It is usually good practice in Python to allow your function inputs to be as broad as possible and your return to be as specific as possible.
That can make sense, but not in all cases. I think this is one of those cases when you want to be more specific.

> I fear that by doing this, we would force Qubit objects onto `get_reduced_ket`.
That's it, but I think that's a good thing in this case.

> I think a ticket is good, this is not the only function where this functionality should be instilled.
I think that's going too far, really.